src/pkg/packager/actions.go

```diff
@@ -200,6 +200,16 @@ func actionCmdMutation(cmd string, shellPref types.ZarfComponentActionShell) (st
 		return cmd, err
 	}
 
+	// if another binary is calling Zarf, use the system Zarf
+	// See https://regex101.com/r/O4Ngia/1
+	regex, err := regexp.Compile(`^.*zarf(\.exe)?$`)
```

Comment:
This will not work for `zarf_v0.28.4_Darwin_all ` - we should probably do something where we try to get package information from the binary.